### PR TITLE
feat: remove billing page, add faq to pricing page and fix navbar

### PR DESCRIPTION
**Pull request by tobySolutions**
## Why?

This PR fixes navbar issues on mobile screens that the website currently has. It also deprecates the billing pages, with all the content from there moved to an FAQ section on the pricing page.

### Pricing page update

https://github.com/user-attachments/assets/5976cbac-58f8-4fea-9acc-bf73d19b5760

### Nabvar fix
https://github.com/user-attachments/assets/b1b553ca-2666-4c76-9025-798cf5d8da84

## Tickets?

- [Move billing FAQs from the Support Center to the /pricing page under the pricing table](https://linear.app/fleekxyz/issue/PLAT-1751/move-billing-faqs-from-the-support-center-to-the-pricing-page-under)

## Contribution checklist?

- [x] The commit messages are detailed
- [x] The `build` command runs locally
- [x] Assets or static content are linked and stored in the project
- [x] Document filename is named after the slug
- [x] You've reviewed spelling using a grammar checker
- [x] For documentation, guides or references, you've tested the commands and steps
- [x] You've done enough research before writing

## Security checklist?

- [x] Sensitive data has been identified and is being protected properly
- [x] Injection has been prevented (parameterized queries, no eval or system calls)
- [x] The Components are escaping output (to prevent XSS)

### CODE REVIEW

File: src/settings.json [hunk 1]
```diff
@@ -401,5 +390,39 @@
         "title": "Submit a request"
       }
     ]
-  }
+  },
+  "faqs": [
```
Comment (heldrida) [inline src/settings.json:394]: The settings.json is not meant to hold content like text. You might get confused, as metadata contains user-friendly text. Place the data next to the component `src/components/Pricing/index.tsx`, e.g. `src/components/Pricing/data.json`.

Once you put the content in the right location, remove need for this "ContentWithLinks". Prefer to use placeholders and find and replace it:

For example `data.json`

```json
{
  "content: "My text contains a link to $newRequests"
}
```

Where `$supportUrl` is a placeholder and must be replaced in the component by the support URL found in settings. If placeholders start with "$", we can assume that settings have the related resource and replace it, e.g. settings.resources['newRequests']

File: src/components/ContentWithLinks/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const ContentWithLinks: React.FC<{ content: string }> = ({ content }) => {
```
Comment (heldrida) [inline src/components/ContentWithLinks/index.tsx:1]: This looks suspicious. Check the comment regarding settings "faqs".

**Review by heldrida — CHANGES_REQUESTED** [review]

File: src/components/ContentWithLinks/index.tsx (hunk 2 above)
Comment (tobySolutions) [reply src/components/ContentWithLinks/index.tsx:1]: Alright, awesome. Thank you. On it now.

File: src/components/Pricing/index.tsx [hunk 3]
```diff
@@ -3,6 +3,25 @@ import TableDesktop from './Table/TableDesktop';
...
+    return resources[key] || match;
+  });
+}
+
+const resources = settings.support.resources || {};
+
+const processedFaqs = data.faqs.map((faq) => ({
+  ...faq,
+  content: replacePlaceholders(faq.content, resources),
+}));
```
Comment (heldrida) [inline src/components/Pricing/index.tsx:24]: If you're trying to replace just the content key value

```suggestion
const processedFaqs = data.faqs.map(({label, content}) => ({
  label,
  content: replacePlaceholders(content, resources),
}));
```

File: src/components/Pricing/index.tsx [hunk 4]
```diff
@@ -3,6 +3,25 @@ import TableDesktop from './Table/TableDesktop';
 import { PricingInfo } from '../../content/pricing/config';
 import PricingCard from '@components/PricingCard';
 import { Text } from '@components/LandingPage/Text';
+import Accordion from '@components/Accordion';
+import settings from '@base/settings.json';
+import data from './data.json';
+
+function replacePlaceholders(
```
Comment (heldrida) [inline src/components/Pricing/index.tsx:10]: Rename to make it easier to understand what it does, e.g. `replaceTextPlaceholderLinks` and move to `src/utils`.

This doesn't create the links, though. The previous wrapped it in link, which's missing here.

File: src/utils/splitContentIntoElements.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+interface Resources {
+  [key: string]: string;
+}
+
+export function splitContentIntoElements(
```
Comment (heldrida) [inline src/utils/splitContentIntoElements.tsx:5]: We know that the utility is only used to replace link placeholders.

```suggestion
export function parseLinkPlaceholderContent(
```

**Review by heldrida — APPROVED** [review]

**Pull request merged**